src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/HaFlowUpdateHubBolt.java

```diff
@@ -86,21 +90,23 @@ protected void handleInput(Tuple input) throws PipelineException {
                 unhandledInput(input);
             }
         } catch (FlowProcessingException e) {
-            emitErrorMessage(e.getErrorType(), COMMON_ERROR_MESSAGE,  e.getMessage());
+            emitErrorMessage(e.getErrorType(), COMMON_ERROR_MESSAGE, e.getMessage());
+        } catch (SwitchNotFoundException e) {
+            emitErrorMessage(ErrorType.NOT_FOUND, COMMON_ERROR_MESSAGE, e.getMessage());
         } catch (Exception e) {
             emitErrorMessage(ErrorType.INTERNAL_ERROR, COMMON_ERROR_MESSAGE, e.getMessage());
         }
     }
 
-    private void handleHaFlowUpdate(String key, HaFlowRequest payload) {
+    private void handleHaFlowUpdate(String key, HaFlowRequest payload) throws SwitchNotFoundException {
         HaFlow returnHaFlow = persistenceManager.getTransactionManager().doInTransaction(() -> {
             Optional<HaFlow> foundHaFlow = haFlowRepository.findById(payload.getHaFlowId());
             if (!foundHaFlow.isPresent()) {
                 throw new FlowProcessingException(ErrorType.NOT_FOUND,
                         format("HA-flow %s not found.", payload.getHaFlowId()));
             }
             HaFlow haFlow = foundHaFlow.get();
-            Map<String, HaSubFlow> subFlowMap = haFlow.getSubFlows().stream()
+            Map<String, HaSubFlow> subFlowMap = haFlow.getHaSubFlows().stream()
```

Comment:
But that's good you can choose from multiple suggestions and we can understand the issue better.
I think HA is needed there because ha subflow and subflow are different java types.